CMakeLists.txt

```diff
@@ -103,49 +103,11 @@ ELSE()
   FIND_PACKAGE(Threads REQUIRED)
   ADD_CXX_FLAG("-std=c++11")
   if (NOT EMSCRIPTEN)
-    # try to get the target architecture by compiling a dummy.c file and
-    # checking the architecture using the file command.
-    file(WRITE ${PROJECT_BINARY_DIR}/dummy.c "main(){}")
-    try_compile(
-      COMPILE_OK
-      ${PROJECT_BINARY_DIR}
-      ${PROJECT_BINARY_DIR}/dummy.c
-      OUTPUT_VARIABLE COMPILE_OUTPUT
-      COPY_FILE ${PROJECT_BINARY_DIR}/dummy
-    )
-    if (COMPILE_OK)
-      execute_process(
-        COMMAND file ${PROJECT_BINARY_DIR}/dummy
-        RESULT_VARIABLE FILE_RESULT
-        OUTPUT_VARIABLE FILE_OUTPUT
-        ERROR_QUIET
-      )
-
-      if (FILE_RESULT EQUAL 0)
-        if (${FILE_OUTPUT} MATCHES "x86[-_]64")
-          set(TARGET_ARCH "x86-64")
-        elseif (${FILE_OUTPUT} MATCHES "Intel 80386")
-          set(TARGET_ARCH "i386")
-        elseif (${FILE_OUTPUT} MATCHES "ARM")
-          set(TARGET_ARCH "ARM")
-        else ()
-          message(WARNING "Unknown target architecture!")
-        endif ()
-        if(TARGET_ARCH)
-          MESSAGE(STATUS "Building for platform ${TARGET_ARCH}")
-        endif ()
-      else ()
-        message(WARNING "Error running file on dummy executable")
-      endif ()
-    else ()
-      message(WARNING "Error compiling dummy.c file: ${COMPILE_OUTPUT}")
-    endif ()
-
-    if (TARGET_ARCH STREQUAL "i386")
+    if (CMAKE_SYSTEM_PROCESSOR MATCHES "^i.86$")
```

Comment:
`i` is from `i386`, `i486`, `i586`, `i686` while `^i` is to avoid matching something with `86` substring that isn't a descendant of Intel 8086. CMake itself uses`^i.86$` pattern in one of its modules.
https://github.com/Kitware/CMake/blob/v3.10.2/Modules/FindJNI.cmake#L40

Reply:
This is a regular expression match, `i` is first character (so `i686` etc).

Reply:
I'm confused then, what about `x86_64`? (which is what it seems it should be on my machine, based on the [cmake docs](https://cmake.org/cmake/help/v3.0/variable/CMAKE_SYSTEM_PROCESSOR.html))

Reply:
It doesn't need special flags. IIRC, this section is just because wasm requires IEEE754 behavior that is provided by SSE2, but not by x87, which will be used by default on i686 builds. x86_64 uses SSE2 by default.

Reply:
Thanks, now I get it :)